### ISSUE TITLE
ora_asm_diskgroup issues

### ISSUE THREAD

**Issue by biemond** [hajee/oracle#33]
Hi Bert,

ora_asm_diskgroup does not work for external redundancy and puppet resource displays wrong values

example in https://github.com/biemond/biemond-oradb-vagrant-11.2-ASM 

    ora_asm_diskgroup { 'RECO@+ASM':
      ensure          => 'present',
      au_size         => '1',
      compat_asm      => '11.2.0.0.0',
      compat_rdbms    => '11.2.0.0.0',
      diskgroup_state => 'MOUNTED',
      failgroups      => {'RECO_0000' => {'diskname' => 'RECO_0000', 'path' => '/nfs_client/asm_sda_nfs_b3'},
                          'RECO_0001' => {'diskname' => 'RECO_0001', 'path' => '/nfs_client/asm_sda_nfs_b4'}
                         },
      redundancy_type => 'EXTERNAL',
      require         => Oradb::Opatch['19791420_grid_patch'],
    }

Issue 1 
in case of external redundancy you can't use failgroup parameter 

this works for external redundancy , other can use the original one 

		create diskgroup <%= groupname -%>
		  <%= redundancy_type -%> redundancy
		  <% failgroups.each do | key, group| -%>
		    disk '<%= group['path'] -%>' name <%= group['diskname'] -%>
		  <% end %>
		  ATTRIBUTE
		    'compatible.asm' = '<%= compat_asm -%>',
		    'compatible.rdbms' = '<%= compat_rdbms -%>';

or this 

         asmca -silent -createDiskGroup -diskGroupName RECO -disk '/nfs_client/asm_sda_nfs_b3' -disk  '/nfs_client/asm_sda_nfs_b4' -redundancy EXTERNAL -au_size 1 -compatible.asm '11.2.0.0.0' -compatible.rdbms '11.2.0.0.0' -compatible.advm '11.2.0.0.0'

EXTERNAL REDUNDANCY indicates that Automatic Storage Management does not provide any redundancy for the disk group. The disks within the disk group must provide redundancy (for example, using a storage array), or you must be willing to tolerate loss of the disk group if a disk fails (for example, in a test environment). You cannot specify the FAILGROUP clause if you specify EXTERNAL REDUNDANCY.

Issue 2
redundancy_type property displays the wrong value with puppet resource ora_asm_diskgroup 
it displays redundancy_type => 'EXTERN'
should be redundancy_type => 'EXTERNAL'

Issue 3
this redundancy_type parameter should be restricted to EXTERNAL , NORMAL & HIGH

Issue 4
au_size property displays the wrong value with puppet resource ora_asm_diskgroup 
it displays au_size  => '1048576'  instead of 1

Issue 5
failgroups parameter should be called disks, this is a better name

**Comment by cliles** [comment]
I ended up using the following template
```
create diskgroup <%= groupname -%>
  <%= redundancy_type -%> redundancy
  <% failgroups.each do | key, group| -%>
    <% if redundancy_type != 'EXTERNAL' -%> failgroup <%= key -%> <% end -%>
    disk '<%= group['path'] -%>' name <%= group['diskname'] -%>
  <% end %>
  ATTRIBUTE
    'compatible.asm' = '<%= compat_asm -%>',
    'compatible.rdbms' = '<%= compat_rdbms -%>';

**Comment by hajee** [comment]
Thanx. If you create a pull request, I will merg it and you get the credit for the work

**Comment by biemond** [comment]
this is working, failgroups.each should be disks.each

    create diskgroup <%= groupname -%>
      <%= redundancy_type -%> redundancy
      <% disks.each do | key, group| -%>
        <% if redundancy_type != 'EXTERNAL' -%>failgroup <%= key -%> <% end -%>
        disk '<%= group['path'] -%>' name <%= group['diskname'] -%>
      <% end %>
      ATTRIBUTE
        'compatible.asm' = '<%= compat_asm -%>',
        'compatible.rdbms' = '<%= compat_rdbms -%>';

sudo puppet resource ora_asm_diskgroup

    ora_asm_diskgroup { 'DATA@+ASM':
      ensure          => 'present',
      au_size         => '1',
      compat_asm      => '11.2.0.0.0',
      compat_rdbms    => '10.1.0.0.0',
      diskgroup_state => 'MOUNTED',
      disks           => {'DATA_0000' => {'diskname' => 'DATA_0000', 'path' => '/nfs_client/asm_sda_nfs_b1'}, 'DATA_0001' => {'diskname' => 'DATA_0001', 'path' => '/nfs_client/asm_sda_nfs_b2'}},
      redundancy_type => 'EXTERNAL',
    }
    ora_asm_diskgroup { 'RECO@+ASM':
      ensure          => 'present',
      au_size         => '1',
      compat_asm      => '11.2.0.0.0',
      compat_rdbms    => '10.1.0.0.0',
      diskgroup_state => 'MOUNTED',
      disks           => {'RECO_0000' => {'diskname' => 'RECO_0000', 'path' => '/nfs_client/asm_sda_nfs_b3'}, 'RECO_0001' => {'diskname' => 'RECO_0001', 'path' => '/nfs_client/asm_sda_nfs_b4'}},
      redundancy_type => 'EXTERNAL',
    }

**Comment by hajee** [comment]
Fixed in b1d8d937d705853385b4ddcdaf299a77c7d309cd
